lib/Data/Validate/Sanctions.pm

```diff
@@ -208,30 +208,27 @@ sub get_sanctioned_info {    ## no critic (RequireArgUnpacking)
 
     $self->_load_data();
 
-    # Sub to remove non-alphabets from the name
-    my $clean_names = sub {
-
-        my ($full_name) = @_;
-
-        # Remove non-alphabets
-        my @cleaned_full_name = split " ", uc($full_name =~ s/[^[:alpha:]\s]//gr);
-
-        return @cleaned_full_name;
-    };
-
     my $client_full_name = join(' ', $first_name, $last_name || ());
 
     # Split into tokens after cleaning
-    my @client_name_tokens = $clean_names->($client_full_name);
+    my @client_name_tokens = _clean_names($client_full_name);
 
     my @match_with_dob_text;
 
-    my %index = $self->{_index}->%*;
-    foreach my $sanctioned_name (sort keys %index) {
-        my @sanctioned_name_tokens = $clean_names->($sanctioned_name);
-        next unless _name_matches(\@client_name_tokens, \@sanctioned_name_tokens);
+    # only pick the sanctioned names which have common token with the client tokens
+    # and deduplicate the list
+    my $filtered_sanctioned_names = {};
+    foreach my $token (@client_name_tokens) {
+        foreach my $name ( keys %{$self->{_token_sanctioned_names}->{$token}}) {
+            $filtered_sanctioned_names->{$name} = 1;
+        }
+    }
+
+    foreach my $sanctioned_name (sort keys %{$filtered_sanctioned_names}) {
```

Comment:
Is sorting needed?

Reply:
@billmarriott-binary no I commented about it above.
@mat-fs can you please adjust the unit test and remove this sort?

Reply:
A card is created for removing the `sort` calls: https://redmine.deriv.cloud/issues/68219#Remove_redundant_sorts_from_sanction_checks

Reply:
fixed here https://github.com/binary-com/perl-Data-Validate-Sanctions/pull/42